File: android/app/src/main/java/com/furnit/android/util/DisplayNameValidation.kt
package com.furnit.android.util

/** Shared rule for login name and room name: ≥3 characters with at least one letter. */
object DisplayNameValidation {
    fun isValid(raw: String): Boolean {
        val trimmed = raw.trim()
        if (trimmed.length < 3) return false
        return trimmed.any { it.isLetter() }
    }
}

File: Furnit/Utilities/DisplayNameValidation.swift
import Foundation

/// Shared rule for login name and room name: ≥3 characters with at least one letter.
enum DisplayNameValidation {
    static func isValid(_ raw: String) -> Bool {
        let trimmed = raw.trimmingCharacters(in: .whitespacesAndNewlines)
        guard trimmed.count >= 3 else { return false }
        return trimmed.contains { $0.isLetter }
    }
}
